src/components/VolunteerAutomation/AttendClass.js

```diff
@@ -0,0 +1,333 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import moment from "moment";
+import { actions as classActions } from "../../components/Class/redux/action";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import axios from "axios";
+import Chip from "@mui/material/Chip";
+import { METHODS } from "../../services/api";
+import {
+  Typography,
+  Container,
+  Grid,
+  Box,
+  Button,
+  Card,
+  CardContent,
+  CardActions,
+} from "@mui/material";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogTitle from "@mui/material/DialogTitle";
+import useStyles from "./styles";
+import { dateTimeFormat, TimeLeft } from "../../constant";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+
+function AttendClass({ setDisable }) {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const user = useSelector(({ User }) => User);
+  const { data = [] } = useSelector(({ Class }) => Class.allClasses);
+  const [enrollId, setEnrollId] = useState(false);
+  const [open, setOpen] = React.useState(false);
+  const [proceed, setProceed] = useState(
+    JSON.parse(localStorage.getItem("proceed")) || false
+  );
+  const [chooseClassAgain, setChooseClassAgain] = useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  useEffect(() => {
+    dispatch(classActions.getClasses());
+  }, [dispatch]);
+
+  const languageMap = {
+    hi: "Hindi",
+    te: "Telugu",
+    en: "English",
+    ta: "Tamil",
+    cohort: "Batch",
+  };
+
+  const classData = JSON.parse(localStorage.getItem("classes"));
+
+  let sliceData = classData || [];
+  chooseClassAgain &&
+    data &&
+    data.slice(0, 6).map((item) => {
+      sliceData.push(item);
+    });
+
+  !chooseClassAgain &&
+    sliceData &&
+    sliceData.find((item) => {
+      if (item.id == enrollId) {
+        sliceData = [];
+        sliceData.push(item);
+        localStorage.setItem("classes", JSON.stringify(sliceData));
+      }
+    });
```

Comment:
`find` shouldn't be used to build up an Array, it should be used to get the first item in an `Array` that meets a certain condition, but it looks as if that's what you're trying to do as you only want to get one enrolled class (id)?  If so, you could do something like the following:
```
const enrolledClass = !chooseClassAgain && sliceData.find((item) => item.id === "enrolled");
enrolledClass && localStorage.setItem("enrolledClass", enrolledClass); 
```
But again, we'll want to bundle the state so I'll address this in my PR.